tools/testing/test_selections.py

```diff
@@ -195,6 +196,67 @@ def _python_test_file_to_test_name(tests: Set[str]) -> Set[str]:
     return valid_tests
 
 
+class PoolTimes:
+    def __init__(self, num_procs: int) -> None:
+        self.pool_times = [0.0 for _ in range(num_procs)]
+        self.serial_times = 0.0
+
+    def next_test_start_time(self, serial: bool) -> float:
+        if serial:
+            # Serial tests are run after all parallel tests complete
+            return max(self.pool_times) + self.serial_times
+
+        return self.pool_times[0]
+
+    def schedule_test(self, test: ShardedTest, serial: bool) -> None:
+        if serial:
+            self.serial_times += test.get_time()
+        else:
+            # pool_times[0] is always the thread with the least amount of time scheduled
+            heapq.heappushpop(self.pool_times, self.pool_times[0] + test.get_time())
+
+
+def log_time_savings(
+    selected_tests: List[ShardedTest],
+    prioritized_tests: List[ShardedTest],
+    is_serial_test_fn: Callable[[str], bool],
+    num_procs: int = NUM_PROCS,  # make this customizable for testing
+) -> float:
+    # The tests will be run in [num_procs] parallel threads, so we assume each test
+    # is allocated to the thread that'll free up first.
+    # This isn't an exact match (since other factors could change which thread
+    # pool a test gets scheduled on) but it's a good approximation.
+
+    # Simulates the scheduled tests on each thread pool
+    default_pool = PoolTimes(num_procs)  # originally scheduled run
+    prioritized_pool = PoolTimes(num_procs)  # run for prioritized tests
+    max_time_savings_sec = 0.0
+
+    # It's easier to look up prioritized tests by name
+    prioritized_test_names = {test.name for test in prioritized_tests}
+
+    for test in selected_tests:
+        serial = is_serial_test_fn(test.name)
+        if test.name in prioritized_test_names:
+            # Successive tests will always have a greater time savings
+            max_time_savings_sec = default_pool.next_test_start_time(
```

Comment:
Should there also be a max here as the name implies, i.e. max_time_savings_sec = max(max_time_savings_sec, default_pool.next_test_start_time(serial) - prioritized_pool.next_test_start_time(serial)`

Reply:
Got it!  I should have pay more attention to comments when reviewing code.  I try to read the code to understand it as it goes, then map my review questions accordingly.  I didn't even notice the comment until you point it out explicitly (blind spot, lol)